### PR TITLE
feat(cellguide): add search-by-synonyms

### PR DESCRIPTION
**Pull request by atarashansky**
## Reason for Change

- Unblocks #5208 
- Search by synonyms was a requirement from the beginning but it flew under the radar until it was required for SEO. So I'm adding that functionality here!

## Testing steps

- Added an E2E test to search by synonym
- tested locally

### CODE REVIEW

File: cell_guide_local_pipeline/cell_guide_data_pipeline.ipynb [hunk 1]
```diff
@@ -83,7 +83,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "sn = \"prod-snapshot\"\n",
+    "sn = \"staging-snapshot\"\n",
```
Comment (tihuan) [inline cell_guide_local_pipeline/cell_guide_data_pipeline.ipynb:86]: is the change to staging snapshot intentional here?

File: cell_guide_local_pipeline/cell_guide_data_pipeline.ipynb [hunk 2]
```diff
@@ -106,7 +106,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 5,
+   "execution_count": 4,
```
Comment (tihuan) [inline cell_guide_local_pipeline/cell_guide_data_pipeline.ipynb:109]: why is this down by 1?

**Review by tihuan — APPROVED** [review]
LGTM! Just some small non blocking comments 🎉 Thanks for adding this, Alec 💯

File: cell_guide_local_pipeline/cell_guide_data_pipeline.ipynb (hunk 1 above)
Comment (atarashansky) [reply cell_guide_local_pipeline/cell_guide_data_pipeline.ipynb:86]: we can ignore changes to the notebook. ty for pointing this out though!

File: cell_guide_local_pipeline/cell_guide_data_pipeline.ipynb (hunk 2 above)
Comment (atarashansky) [reply cell_guide_local_pipeline/cell_guide_data_pipeline.ipynb:109]: just jupyter notebook things...

**Pull request merged**